src/Select/web/Select.js

```diff
@@ -172,40 +178,46 @@ class Select extends React.Component {
                   <OptionList
                     isOpen={isOpen}
                     block={block}
+                    height={optionsListHeight}
                   >
-                    {
-                      options
-                        .map(this.makeOption)
-                        .map((option, index) => (
-                          <Option
-                            {...getItemProps({
-                              key: option.value,
+                    <VirtualList
+                      width="100%"
```

Comment:
Once you remove the min-width(as mentioned above), you'll have to implement the condition for `block`

```
width={block ? '100%' : px(25)
```